.github/actions/check-changelog/entrypoint.sh

```diff
@@ -3,17 +3,32 @@
 # Exit on error
 set -e
 
-# The PR number provided as input
+# The PR number and package path provided as input
 PR_NUMBER="${PR_NUMBER}"
+PACKAGE_PATH="${PACKAGE_PATH}"
 
-# Search for the changelog file across all packages
-changelog_files=$(find packages/*/changelogs/upcoming/ -type f -name "${PR_NUMBER}.md")
+# Search for the changelog file
+if [ -n "$PACKAGE_PATH" ]; then
+  # Check specific package
+  changelog_files=$(find "${PACKAGE_PATH}/changelogs/upcoming/" -type f -name "${PR_NUMBER}.md" 2>/dev/null || true)
+  package_name=$(basename "$PACKAGE_PATH")
 
-# If no changelog file is found, fail the step with an error message
-if [ -z "$changelog_files" ]; then
-  echo "❌ Changelog file for PR #${PR_NUMBER} is missing."
-  echo "You need to add a changelog to this PR before it can be merged. See https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md."
-  exit 1
+  if [ -z "$changelog_files" ]; then
+    echo "❌ Changelog file for PR #${PR_NUMBER} is missing in package '${package_name}'."
+    echo "You need to add a changelog to this PR before it can be merged. See https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md."
+    exit 1
+  else
+    echo "✅ Changelog file for PR #${PR_NUMBER} found in package '${package_name}': $changelog_files"
+  fi
 else
-  echo "✅ Changelog file for PR #${PR_NUMBER} found: $changelog_files"
+  # Search for the changelog file across all packages
+  changelog_files=$(find packages/*/changelogs/upcoming/ -type f -name "${PR_NUMBER}.md")
+
+  if [ -z "$changelog_files" ]; then
+    echo "❌ Changelog file for PR #${PR_NUMBER} is missing."
+    echo "You need to add a changelog to this PR before it can be merged. See https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md."
```

Comment:
```suggestion
    echo "You need to add a changelog to this PR before it can be merged. See https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md"
```

[source](https://github.com/elastic/eui/pull/9054#discussion_r2378534339)